mbedtls-sys/build/cmake.rs

```diff
@@ -32,7 +32,9 @@ impl super::BuildConfig {
         // thumbv6m-none-eabi, thumbv7em-none-eabi, thumbv7em-none-eabihf,
         // thumbv7m-none-eabi probably use arm-none-eabi-gcc which can cause the
         // cmake compiler test to fail.
-        if target.starts_with("thumbv") && target.contains("none-eabi") {
+        if target.starts_with("thumbv") && target.contains("none-eabi")
+            || crate::features::FEATURES.have_platform_component("c_compiler", "freestanding")
```

Comment:
This will affect the environment which already has `freestanding` feature.
I think you could just directly use read `RUST_MBED_C_COMPILER_FREESTANDING` and use it here. And it could have another name, such as `RUST_MBED_C_COMPILER_ARM_BAREMETAL` or anything more related to the issue or reason here.


In `mbedtls-sys/build/features.rs` you do not need to change the logic about adding `freestanding` feature.

Reply:
Also, I am thinking if your case can be determined by checking some property of `cc` here.

Note: `cc` is from `let cc = cc::Build::new().get_compiler();`

Reply:
Querying from CC could work but is not guaranteed for example we can check for `-none` targets however there might be other operating systems that purely only support static libraries.

I've tested 30b7f134339c334f47563437c26fa9327a24b656 with 
```sh
$ CFLAGS="-isysroot /opt/arm-gnu-toolchain-13.2.Rel1-x86_64-aarch64-none-elf" CC=clang RUST_MBED_C_COMPILER_BAREMETAL=1 cargo +stable b --target aarch64-unknown-none --no-default-features --features no_std_deps
```